src/test/java/com/aws/greengrass/logging/impl/StructuredLayoutTest.java

```diff
@@ -74,4 +78,34 @@ public void GIVEN_text_appender_WHEN_write_log_events_THEN_events_are_patterned_
             assertEquals(message.getTextMessage(), appendedMessages[i]);
         }
     }
+
+    @Test
+    public void GIVEN_throwable_that_fails_default_serializer_WHEN_log_it_THEN_custom_serializer_succeeds()
+            throws JsonProcessingException {
+        try {
+            // This will throw an exception that cannot be serialized successfully by Jackson's default serializer
+            JSON_MAPPER.readValue("this will go wrong", Object.class);
+            fail("Expected to raise exception");
+        } catch (Throwable e) {
+            try {
+                JSON_MAPPER.writeValueAsString(e);
+                fail("Expected default serializer to fail");
+            } catch (Exception ignore) {
+                // do nothing
+            }
+            GreengrassLogMessage message = new GreengrassLogMessage();
+            message.setMessage("testing");
+            message.setCause(e);
+            GreengrassLogMessage messageDeserialized =
+                    JSON_MAPPER.readValue(message.getJSONMessage(), GreengrassLogMessage.class);
+            assertEquals(message, messageDeserialized);
+
+            Throwable deserializedCause = messageDeserialized.getCause();
+            assertEquals(e.getMessage(), deserializedCause.getMessage());
```

Comment:
validate that no additional fields (not present in a regular `Throwable`) get serialized out from either the exception, a suppressed exception, or a cause.